imap_processing/idex/idex_packet_parser.py

```
@@ -89,6 +93,48 @@ def __init__(self, packet_file: str):
 
         self.data = xr.concat(processed_dust_impact_list, dim="Epoch")
 
+    def write_l1_cdf(self, description: str = "", version: str = "01"):
+        """Write the contents of self.data to a CDF file.
+
+        The date in the file name is determined by the start time of the first event
+        in the file
+
+        Parameters
+        ----------
+            description (str): The description to insert into the file name after the
+                               orbit, before the SPICE field.  No underscores allowed.
+            version (str):  The version number to append to the file
+
+        Returns
+        -------
+            str
+                The name of the file created
+        """
+        # Set the global attributes in the xarray
+        self.data.attrs = idex_cdf_attrs.idex_l1_global_attrs
+
+        # Determine the start date of the data in the file,
+        # based on the time of the first dust impact
+        file_start_date = self.data["Epoch"][0].data
+        date_string = np.datetime_as_string(file_start_date, unit="D").replace("-", "")
+        if description:  # Just append an underscore to the description
+            description = f"_{description}"
+        # Set file name for IDEX L1 based on the date string above and the version
+        filename = f"imap_idx_l1_{date_string}_0000{description}_p_v{version}.cdf"
+
+        # Convert the xarray object to a CDF!
+
+        try:
+            xarray_to_cdf(
+                self.data, filename, terminate_on_warning=True
+            )  # Terminate if not ISTP compliant
+        except Exception as e:
+            logging.error("Error writing a CDF file! Aborting....")
+            logging.error(str(e))
+            return
```

Why not just let this raise? I don't think we need this in a try/except. You say terminate, but then you just return from the exception instead of re-raising.

---

I'll change that. I initially just figured if we wanted to write multiple IDEX files we didn't want the whole program to crash.  But we can just have the try/except located outside of whatever calls this function in that case.